sdk/resource/auto.go

```diff
@@ -49,31 +51,64 @@ func Detect(ctx context.Context, detectors ...Detector) (*Resource, error) {
 	return r, detect(ctx, r, detectors)
 }
 
+type detectInternalResult struct {
+	detectorIndex int // index in the detector slice
+	resource      *Resource
+	err           error
+}
+
+func detectInternal(ctx context.Context, detectors []Detector) []detectInternalResult {
+	wg := sync.WaitGroup{}
+	mu := sync.Mutex{} //mu guard resultSlice during goroutines execution
+	resultSlice := make([]detectInternalResult, 0, len(detectors))
```

Comment:
Does this get allocated to the stack if `detectInternal` is inlined in `Detect`?

Reply:
on my branch
./auto.go:61:2: moved to heap: wg
./auto.go:62:2: moved to heap: mu
./auto.go:63:2: moved to heap: resultSlice
./auto.go:63:21: make([]detectInternalResult, 0, len(detectors)) escapes to heap


after inlining it manually

./auto.go:75:2: moved to heap: wg
./auto.go:76:2: moved to heap: mu
./auto.go:77:2: moved to heap: results
./auto.go:77:17: make([]detectInternalResult, 0, len(detectors)) escapes to heap

this will just make code more un-readable and not change performance at all

again performance of this function is not **THAT** critical compared for example to the trace creation code
it run only once at initialization of resource at main function 
or i got that part wrong?